client/ayon_substancepainter/plugins/publish/collect_textureset_images.py

```diff
@@ -209,9 +209,18 @@ def get_export_config(self, instance):
         }
 
         # Create the list of Texture Sets to export.
-        config["exportList"] = []
-        for texture_set in substance_painter.textureset.all_texture_sets():
-            config["exportList"].append({"rootPath": texture_set.name()})
+        export_texture_sets = creator_attrs.get("exportTextureSets", [])
+        if export_texture_sets:
+            config["exportList"] = [
+                {"rootPath": texture_set_name}
+                for texture_set_name in export_texture_sets
+            ]
+        else:
+            config["exportList"] = [
+                {"rootPath": texture_set_name}
+                for texture_set_name in
+                substance_painter.textureset.all_texture_sets()
+            ]
```

Comment:
```suggestion
        if not export_texture_sets:
            # Export all texture sets
            export_texture_sets = {
                texture_set.name() for texture_set in
                substance_painter.textureset.all_texture_sets()
            }
        config["exportList"] = [
            {"rootPath": texture_set_name}
            for texture_set_name in export_texture_sets
        ]
```

However, I still think this is the wrong **fix** to "export all". Didn't it work to just **NOT** write `config["exportList"]` key at all so that we provide no override whatsoever to the export list when empty?

The code before this 'fix' https://github.com/ynput/ayon-substance-painter/commit/4d3fbaf4bf055f8a8fc336ffe8f236e0f9c1a4db was actually still setting `config["exportList"] = []` which makes sense that it wouldn't export anything... but if we provide NO `exportList` alltogether - shouldn't it just export without any filtering applied?